### ISSUE TITLE
Compilation database not always found

### ISSUE THREAD

**Issue by matthiasvegh** [bbchung/clighter#27]
On one of my projects, compile_commands.json is found correctly and has done for quite some time now.
At work one of my projects the compilation database seems to have been found correctly in the past (or atleast highlighting worked as expected). However on the current master highlighting is missing, and ClighterShowArgs shows None as the compilation database.

I did a bisect to find the first faulty commit (on which the work projects' compile_commands.json is no longer detected) and it is bc724a234eac91986362d4b1042dc11a1e12b9d2.

My compile_commands.json is in the root of the project, one of the files where I notice this is: project_root/source/server/net/thrift/DasThriftSerialization/src/Metadata.cpp

An extract from my compile_commands.json is:

    { "directory": "/home/emtyvgh/workspace/das",                                                                                                                                                                                                                                                                                
    "command": "clang++ -O2 -pipe -D__unittest__ -D__cba__ -g -fno-omit-frame-pointer -DDDB_DEBUG -fPIC -DUNIT_IDENTITY=SCC/DasThriftSerialization -DUNIT_VERSION=R1A02 -Wall -Wcast-qual -I <lots of -I include flags...> -MD -c -o /home/emtyvgh/workspace/das/source/server/net/thrift/DasThriftSerialization_SCC/obj/test-unit/Metadata.cpp.o /home/emtyvgh/workspace/das/source/server/net/thrift/DasThriftSerialization_SCC/src/Metadata.cpp",                                                                                                                                                                                                                                                                                                        
    "file": "/home/emtyvgh/workspace/das/source/server/net/thrift/DasThriftSerialization_SCC/src/Metadata.cpp" }, ...

I ran the entire json through a validator to check that it is valid. So I don't think that is the issue.
Any help you could provide would be appreciated, at the moment I have downgraded to 49f8808 as that is the last version that still seems to work.

**Comment by bbchung** [comment]
try launch the vim from the directory with compile_commands.json, clighter only fetch the compile_commands.json in that directory

**Comment by matthiasvegh** [comment]
That is where I'm launching vim from, and as my bisecting showed, before bc724a2 the compilation flags were picked up correctly.

**Comment by bbchung** [comment]
can you give me your entire compilation database to find out the problem?

**Comment by matthiasvegh** [comment]
Sorry I can't (it contains a lot of sensitive info), but I'll try to produce a representative example as time permits.

**Comment by bbchung** [comment]
I use loose check for compilation database in newest commit, check if it can now load the compilation database. If not I will keep debug, if yes, you had better check your compilation database format

**Comment by matthiasvegh** [comment]
Hi, I just checked, and latest master (6a66d79) and it works for me but not because it solved the issue. I recreated my compile_commands.json (as I'm not using cmake, I have to generate it, which of course is error-prone). And now it works both with latest master and with the version one before (8702609), this demonstrates that it was my compile_commands.json that was faulty.
I did a diff on what changed, and it turns out that if there are files with empty commands (ie.: command = "") then the json is not successfully applied.
In my case, I am opening Metadata.cpp, and some other file (say foobar.cpp) has an empty command string, then the json is not applied to any source files at all.
While cmake would never generate such a compile_commands.json, in the case of compile_commands.json file from other sources, it might be helpful if clighter could tolerate this, or atleast document this in the troubleshooting portion of the README.
